### PR TITLE
Update compatibility rules

### PR DESCRIPTION
**Pull request by wurzka**

### CODE REVIEW

**Comment by wurzka** [comment]
I was very unsure how to formulate this so that it is understandable what we want to say.

File: docs/json.md [hunk 1]
```diff
@@ -25,14 +25,17 @@ If a change to the format is considered incompatible, then the `formatVersion` h
```
Comment (albertmink) [inline docs/json.md:25]:
```suggestion
For compatible changes to the file format, the format is just updated.
The indicator whether the format has to be changed should not be updated.
```

**Review by albertmink — APPROVED** [review]
Content wise, it think it is an improvement. I noted that we say something about the compatibility in two places. Maybe it is worth to clean it up (here or in an other PR). I see two options for where we can say the compatibility thing:
* Separately in the section `Compatible File Format Changes` and `Incompatible File Format Changes`.
* Put the compatibility thing in the above section.

**Comment by wurzka** [comment]
I would prefer having it separately in the two sections. Nevertheless, it doesn't disturb me that it was already mentioned before that formatVersion is increased in case of an incompatible change.

**Review by schneidermic0 — APPROVED** [review]
LGTM

**Pull request merged**